### ISSUE TITLE
ntp-daemon fails to run

### ISSUE THREAD

**Issue by MikaelUrankar** [pendulum-project/ntpd-rs#1549]
Broken with https://github.com/pendulum-project/ntpd-rs/commit/32f037699b114bc3227e2389ff877ca5ed23a0cb

./target/debug/ntp-daemon -c ntp.toml -l trace
2024-07-12T11:41:58.809389Z  INFO ntpd::daemon: Log level override from command line arguments is active
2024-07-12T11:41:58.809467Z DEBUG ntpd::daemon: Configuration loaded, spawning daemon jobs
Exit 1

This is on FreeBSD, rust 1.79.0

**Comment by MikaelUrankar** [comment]
It seems to fail in ntpd/src/daemon/local_ip_provider.rs, let mut change_listener = ChangeDetector::new()?

**Comment by MikaelUrankar** [comment]
Tracing ntp-daemon show:
```
13723: socket(PF_LOCAL,SOCK_SEQPACKET,0)         = 6 (0x6)
13723: bind(6,{ AF_UNIX "/var/run/devd.seqpacket.pipe" },106) ERR#48 'Address already in use'
```

**Comment by MikaelUrankar** [comment]
It seems to work if I replace "bind" with "connect" in https://github.com/pendulum-project/timestamped-socket/blob/v0.2.1/src/interface/freebsd.rs#L41

**Comment by davidv1992** [comment]
Thank you for the input, the situation should be better with version 1.2.1. Please let us know if there are still issues on freebsd.

**Comment by MikaelUrankar** [comment]
Thanks for the update.
It still fails on FreeBSD, the timestamped-socket version in Cargo.toml is still 0.2.1 (this is with https://github.com/pendulum-project/ntpd-rs/archive/refs/tags/v1.2.1.tar.gz)

**Comment by davidv1992** [comment]
Apologies, I messed up in the 1.2.1 release, 1.2.2 coming soon

**Comment by davidv1992** [comment]
Ok should be fixed now with the 1.2.2 release
